casanovo/casanovo.py

```diff
@@ -489,6 +522,44 @@ def _get_model_weights(cache_dir: Path) -> str:
             )
 
 
+def _resolve_output(
+    output_dir: str | None,
+    output_root: str | None,
+    file_patterns: list[str],
+    verbosity: str,
+) -> Tuple[Path, str]:
+    """
+    Resolves the output directory and sets up logging.
+
+    Parameters:
+    -----------
+    output_dir : str | None
+        The path to the output directory. If `None`, the current working
+        directory will be used.
+    output_root : str | None
+        The base name for the output files. If `None`, no specific base name is
+        set, and logging will be configured accordingly to the behavior of
+        `setup_logging`.
+    file_patterns : list[str]
+        A list of file patterns that should be checked within the `output_dir`.
+    verbosity : str
+        The verbosity level for logging.
+
+    Returns:
+    --------
+    Tuple[Path, str]
+        The output directory and the base name for log and results files (if
+        applicable).
+    """
+    output_dir = Path(output_dir) if output_dir is not None else Path.cwd()
```

Comment:
IMO for a non-existing path it would be best to throw an exception and let the user create the path if that's where they actually wanted to write the output files to, or at the very least pop a warning if a new path is created.

Reply:
I think an exception is maybe a bit excessive fora non-existing path. You could envision running Casanovo in a script with slightly different input every time and writing to different subfolders. Instead, creating the path and issuing a warning seems best.